core/src/main/java/uk/gov/justice/services/core/sender/SenderProducer.java

```diff
@@ -23,19 +24,23 @@
 @ApplicationScoped
 public class SenderProducer {
 
-    @Inject
-    JmsSenderFactory jmsSenderFactory;
 
-    @Inject
-    ComponentDestination componentDestination;
+    private JmsSenderFactory jmsSenderFactory;
+    private ComponentDestination componentDestination;
+    private DispatcherProducer dispatcherProducer;
 
-    @Inject
-    DispatcherProducer dispatcherProducer;
+    private final Map<Component, Sender> senderMap = new ConcurrentHashMap<>();
 
-    private Map<Component, Sender> senderMap;
+    //for CDI proxying only
+    protected SenderProducer() {
```

Comment:
If we go that way then our beans are not POJOs and can't be used outside of the container, e.g. in tests.

I've done some further investigation on bean scopes and constructors and there doesn't seem to be a good solution:
-@ApplicationScoped beans are indeed effectively singletons but require a default constructor
-by default (without any annotations) beans have a so-called dependent pseudo-scope. 
They then don't need default constructors but are not singletons. Each calling bean holds a copy of the dependent bean. Unfortunately this is also true for producers. So SenderProducer bean gets created each time a sender is produced.
-There is also a @Singleton pseudo scope which in theory does not create proxies but still for some reason requires default constructors.